### ISSUE TITLE
Builds failing with MSVC 2012

### ISSUE THREAD

**Issue by clokep** [wojtekka/libgadu#17]
We recently updated Instantbird's libpurple to 2.10.11 and gg is failing to build on Windows with Visual Studio 2012 (I'm using the free edition). It looks like some of the libgadu code is conflicting with (a new?) internal headers from MSVC.

I had [started trying to fix this](https://bugzilla.mozilla.org/attachment.cgi?id=8641023&action=diff), but the changes to `gg_win32_close` feel wrong to me and I was hoping you'd have some idea of what to do. Thanks!

```
 0:56.72 C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\INCLUDE\io.h(326) : error C2375: 'gg_win32_close' : redefinition; different linkage
 0:56.72         c:\users\clokep\comm-central\mozilla\extensions\purple\libpurple\protocols\gg\lib\network.h(68) : see declaration of 'gg_win32_close'
 0:56.73
 0:56.75 In the directory  /c/Users/clokep/comm-central/obj-i686-pc-mingw32/extensions/purple/libpurple/protocols/gg
 0:56.75 The following command failed to execute properly:
 0:56.75 c:/Users/clokep/comm-central/obj-i686-pc-mingw32/_virtualenv/Scripts/python.exe -m mozbuild.action.cl cl -Focommon.obj -c -DHAVE_CONFIG_H -DPACKAGE="gg" -DVERSION="2.10.11" -DDISPLAY_VERSION="2.10.11" -DPURPLE_STATIC_PRPL -DGLIB_STATIC_COMPILATION -Dinline=__inline -DAB_CD=en-US -DNO_NSPR_10_SUPPORT -Ic:/Users/clokep/comm-central/mozilla/extensions/purple/libpurple/protocols/gg -I. -I../../../../../dist/include/libpurple -I../../../../../dist/include/glib -Ic:/Users/clokep/comm-central/mozilla/extensions/purple/libpurple/protocols/gg/lib -I../../../../../dist/include -Ic:/Users/clokep/comm-central/obj-i686-pc-mingw32/dist/include/nspr -Ic:/Users/clokep/comm-central/obj-i686-pc-mingw32/dist/include/nss -MD -FI ../../../../../dist/include/mozilla-config.h -DMOZILLA_CLIENT -TC -nologo -D_HAS_EXCEPTIONS=0 -W3 -Gy -arch:IA32 -FS -wd4244 -wd4267 -wd4819 -we4553 -DNDEBUG -DTRIMMED -Zi -UDEBUG -DNDEBUG -O1 -Oi -Oy -Fdgenerated.pdb c:/Users/clokep/comm-central/mozilla/extensions/purple/libpurple/protocols/gg/lib/common.c
 0:56.75 c:/Users/clokep/comm-central/mozilla/config/rules.mk:918: recipe for target 'common.obj' failed
 0:56.75 mozmake.EXE[4]: *** [common.obj] Error 1
 0:56.75 c:/Users/clokep/comm-central/mozilla/config/recurse.mk:71: recipe for target 'extensions/purple/libpurple/protocols/gg/target' failed
 0:56.75 mozmake.EXE[3]: *** [extensions/purple/libpurple/protocols/gg/target] Error 2
 0:56.76 c:/Users/clokep/comm-central/mozilla/config/recurse.mk:32: recipe for target 'compile' failed
 0:56.76 mozmake.EXE[2]: *** [compile] Error 2
 0:56.76 c:/Users/clokep/comm-central/mozilla/config/recurse.mk:130: recipefor target 'compile' failed
 0:56.76 mozmake.EXE[1]: *** [compile] Error 2
 0:56.76 c:/Users/clokep/comm-central/mozilla/config/rules.mk:549: recipe for target 'default' failed
 0:56.76 mozmake.EXE: *** [default] Error 2
2
```

(Cross-linking the [Instantbird bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1189247))

**Comment by twasilczyk** [comment]
I think io.h should be included before network.h (so the #deine won't touch it). Will look at this later.

**Comment by clokep** [comment]
That would be great! Thanks so much. :)

**Comment by twasilczyk** [comment]
Fixed by af2cc5e. Please verify.

**Comment by clokep** [comment]
Sorry, I've been unable to verify this yet. My Windows machine has been a bit on the fritz. I'll hopefully get to it soon!
